includes/log-handlers/class-wc-log-handler-email.php

```diff
@@ -0,0 +1,118 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Log_Handler' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/abstracts/abstract-wc-log-handler.php' );
+}
+
+/**
+ * Handles log entries by sending an email.
+ *
+ * @class          WC_Log_Handler_Email
+ * @version        1.0.0
+ * @package        WooCommerce/Classes
+ * @category       Class
+ * @author         WooThemes
+ */
+class WC_Log_Handler_Email extends WC_Log_Handler {
+
+	/**
+	 * Stores email recipients.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	private $to = array();
+
+	/**
+	 * Constructor for the logger.
+	 *
+	 * @param $args additional args. {
+	 *     Optional. @see WC_Log_Handler::__construct.
+	 *
+	 *     @type string|arr $to Optional. Email or emails to recieve log messages. Default site admin.
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'threshold' => 'critical',
+			'to' => get_option( 'admin_email' ),
+		) );
+
+		parent::__construct( $args );
+
+		if ( is_array( $args['to'] ) ) {
+			foreach ( $args['to'] as $to ) {
+				$this->add_email( $to );
+			}
+		} else {
+			$this->add_email( $args['to'] );
+		}
+
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message
+	 * @param array $context {
+	 *     Array with additional information
+	 * }
+	 *
+	 * @return bool log entry should bubble to further loggers.
+	 */
+	public function handle( $level, $timestamp, $message, $context ) {
+
+		if ( $this->should_handle( $level ) ) {
+			$subject = $this->get_subject( $level, $timestamp, $message, $context );
+			$body = $this->get_body( $level, $timestamp, $message, $context );
+			wp_mail( $this->to, $subject, $body );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Build subject for log email.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message
+	 * @param array $context {
+	 *     Array with additional information
+	 * }
+	 *
+	 * @return string subject
+	 */
+	public function get_subject( $level, $timestamp, $message, $context ) {
+		return sprintf( __( '[%s] WooCommerce log message', 'woocommerce' ), $level );
+	}
+
+	/**
+	 * Build body for log email.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message
+	 * @param array $context {
+	 *     Array with additional information
+	 * }
+	 *
+	 * @return string body
+	 */
+	public function get_body( $level, $timestamp, $message, $context ) {
+		$entry = $this->format_entry( $level, $timestamp, $message, $context );
+		return sprintf( __( "You have recieved the following WooCommerce log message: \n\n%s", 'woocommerce' ), $entry );
```

Comment:
Best not use "end of line" inside a translatable string, people some times ignore, not to mention that GlotPress transform it into `↵` (what is really funny since if you click in the "Copy from original" will copy this kind of character too).

See one example with `\n` and `\t`:
![screenshot from 2016-11-16 17-50-51](https://cloud.githubusercontent.com/assets/1264099/20363145/5281f69e-ac25-11e6-8f6d-8dc8156a9ec3.png)

Better change to:

`return sprintf( __( "You have recieved the following WooCommerce log message: %s", 'woocommerce' ), PHP_EOL . PHP_EOL . $entry );`

And better prefer `PHP_EOL` since is more safe for Windows and Unix.